### PR TITLE
service-profiles: Prevent Duration coercion panics

### PR DESCRIPTION
**Pull request by olix0r**
When an invalid duration is provided by the control plane, retry budget
conversion can panic. This error is triggered by quickcheck tests.

This change explicitly handles these negative values to avoid panicking.

### CODE REVIEW

**Comment by olix0r** [comment]
Really, it seems like `prost-types` should use checked operations when dealing with network-provided values. We should probably submit fixes upstream...

**Comment by olix0r** [comment]
https://github.com/danburkert/prost/issues/423

File: linkerd/service-profiles/src/client.rs [hunk 1]
```diff
@@ -460,25 +460,37 @@ fn convert_retry_budget(orig: api::RetryBudget) -> Option<Arc<Budget>> {
...
-        Some(pb_dur) => match pb_dur.try_into() {
-            Ok(dur) => {
-                if dur > Duration::from_secs(60) || dur < Duration::from_secs(1) {
-                    warn!("retry_budget ttl invalid: {:?}", dur);
-                    return None;
-                }
-                dur
+        Some(pb_dur) => {
+            if pb_dur.nanos > 1_000_000_000 {
+                warn!("retry_budget nanos must not exceed 1s");
```
Comment (hawkw) [inline linkerd/service-profiles/src/client.rs:473]: nit: should this include the invalid value?

File: linkerd/service-profiles/src/client.rs [hunk 2]
```diff
@@ -460,25 +460,37 @@ fn convert_retry_budget(orig: api::RetryBudget) -> Option<Arc<Budget>> {
...
+                return None;
             }
-            Err(negative) => {
-                warn!("retry_budget ttl negative: {:?}", negative);
+            if pb_dur.seconds < 0 || pb_dur.nanos < 0 {
+                warn!("retry_budget ttl negative");
```
Comment (hawkw) [inline linkerd/service-profiles/src/client.rs:477]: nit: and this one?

**Review by hawkw — APPROVED** [review]

**Pull request merged**